### DISCUSSION TITLE
Re-issuing failed requests

### DISCUSSION THREAD

**Discussion by bra-fsn** [mitmproxy/mitmproxy#5293] [General]
I wonder if this is possible and if so, how:

1. client sends an HTTP request
2. mitmproxy lets it through
3. the request fails (for example connection times out), `error` is called
4. an addon from `error` redirects the request to another server (the same way as `http-redirect-requests.py`, but from `error` instead of `request`)

I tried to set `flow.response = http.Response.make(...)` for a start, but even this doesn't work (and I would need to serve a remote response).
Is this possible to do with the current code?

**Comment by mhils** [comment]
> Is this possible to do with the current code?

In short, no. You could manually make repeated requests in the `request` hook and then set `flow.response` there, but that's a hack at best. It's unlikely that we implement such functionality any time soon as it breaks the contract for our event hooks (one `request` hook is always followed by one (`response` xor `error`) hook).
